Sources/Commands/SymbolGraphExtract.swift

```diff
@@ -1,97 +1,79 @@
     let tool: AbsolutePath
-
-    init(tool: AbsolutePath) {
-        self.tool = tool
+    var skipSynthesizedMembers: Bool = false
+    var minimumAccessLevel: AccessLevel = .public
+    var skipInheritedDocs: Bool = false
+    var includeSPISymbols: Bool = false
+    var prettyPrintOutputJSON: Bool = false
+
+    /// Access control levels.
+    public enum AccessLevel: String, RawRepresentable, CustomStringConvertible, CaseIterable, ExpressibleByArgument {
+        // The cases reflect those found in `include/swift/AST/AttrKind.h` of the swift compiler (at commit 03f55d7bb4204ca54841218eb7cc175ae798e3bd)
+        case `private`, `fileprivate`, `internal`, `public`, `open`
+
+        public var description: String { rawValue }
     }
 
-    public func dumpSymbolGraph(
+    /// Creates a symbol graph for `target` in `outputDirectory` using the build information from `buildPlan`. The `outputDirection` determines how the output from the tool subprocess is handled, and `verbosity` specifies how much console output to ask the tool to emit.
+    public func extractSymbolGraph(
+        target: ResolvedTarget,
         buildPlan: BuildPlan,
-        prettyPrint: Bool,
-        skipSynthesisedMembers: Bool,
-        minimumAccessLevel: AccessLevel,
-        skipInheritedDocs: Bool,
-        includeSPISymbols: Bool
+        outputRedirection: Process.OutputRedirection = .none,
+        verbose: Bool,
+        outputDirectory: AbsolutePath
     ) throws {
         let buildParameters = buildPlan.buildParameters
-        let symbolGraphDirectory = buildPlan.buildParameters.symbolGraph
-        try localFileSystem.createDirectory(symbolGraphDirectory, recursive: true)
-
-        // Run the tool for each target in the root package.
-        let targets = buildPlan.graph.rootPackages.flatMap{ $0.targets }.filter{ $0.type == .library || $0.type == .executable }
-        for target in targets {
-            var args = [String]()
-            args += ["-module-name", target.c99name]
-            args += try buildParameters.targetTripleArgs(for: target)
-
-            args += buildPlan.createAPIToolCommonArgs(includeLibrarySearchPaths: true)
-            args += ["-module-cache-path", buildParameters.moduleCache.pathString]
-
-            args += ["-output-dir", symbolGraphDirectory.pathString]
-
-            if prettyPrint { args.append("-pretty-print") }
-            if skipSynthesisedMembers { args.append("-skip-synthesized-members") }
-            if minimumAccessLevel != SwiftPackageTool.DumpSymbolGraph.defaultMinimumAccessLevel {
-                args += ["-minimum-access-level", minimumAccessLevel.rawValue]
-            }
-            if skipInheritedDocs { args.append("-skip-inherited-docs") }
-            if includeSPISymbols { args.append("-include-spi-symbols") }
-
-            print("-- Emitting symbol graph for", target.name)
-            try runTool(args)
+        try localFileSystem.createDirectory(outputDirectory, recursive: true)
```

Comment:
should this support abstract file system instead of assuming local?

Reply:
Since the tool only works locally there didn't seem to be much point, and could be misleading since only the local file system will work.